Lib/fontmake/instantiator.py

```diff
@@ -0,0 +1,545 @@
+        copy_lib: Mapping[str, Any] = default_font.lib
+
+        # The list of glyphs-not-to-export-and-decompose-where-used-as-a-component is
+        # supposed to be taken from the Designspace when a Designspace is used as the
+        # starting point of the compilation process. It should be exported to all
+        # instance libs, where the ufo2ft compilation functions will pick it up.
+        skip_export_glyphs = designspace.lib.get("public.skipExportGlyphs", [])
+
+        return cls(
+            axis_bounds,
+            copy_feature_text,
+            copy_groups,
+            copy_info,
+            copy_lib,
+            designspace.default.location,
+            designspace.rules,
+            glyph_mutators,
+            glyph_name_to_unicodes,
+            info_mutator,
+            kerning_mutator,
+            round_geometry,
+            skip_export_glyphs,
+            special_axes,
+        )
+
+    def generate_instance(
+        self, instance: designspaceLib.InstanceDescriptor
+    ) -> ufoLib2.Font:
+        """Generate an interpolated instance font object for an
+        InstanceDescriptor."""
+        if anisotropic(instance.location):
+            raise ValueError(
+                f"Instance {instance.familyName}-"
+                f"{instance.styleName}: Anisotropic location "
+                f"{instance.location} not supported by varLib."
+            )
+
+        font = ufoLib2.Font()
+
+        # Instances may leave out locations that match the default source, so merge
+        # default location with the instance's location.
+        location = {**self.default_design_location, **instance.location}
+        location_normalized = varLib.models.normalizeLocation(
+            location, self.axis_bounds
+        )
+
+        # Kerning
+        kerning_instance = self.kerning_mutator.instance_at(location_normalized)
+        if self.round_geometry:
+            kerning_instance.round()
+        kerning_instance.extractKerning(font)
+
+        # Info
+        self._generate_instance_info(instance, location_normalized, location, font)
+
+        # Glyphs
+        for glyph_name, glyph_mutator in self.glyph_mutators.items():
+            glyph = font.newGlyph(glyph_name)
+
+            try:
+                glyph_instance = glyph_mutator.instance_at(location_normalized)
+
+                if self.round_geometry:
+                    glyph_instance = glyph_instance.round()
+
+                # onlyGeometry=True does not set name and unicodes, in ufoLib2 we can't
+                # modify a glyph's name. Copy unicodes from default font.
+                glyph_instance.extractGlyph(glyph, onlyGeometry=True)
+            except Exception as e:
+                # By default, explode if we cannot generate a glyph instance for
+                # whatever reason (usually outline incompatibility)...
+                if glyph_name not in self.skip_export_glyphs:
+                    raise ValueError(
```

Comment:
Maybe it would be good to have our own custom Exception class instead of generic ValueError here